// yandex-dialogs-sdk/dist/reply/reply.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const textReplyBuilder_1 = require("./textReplyBuilder");
const extraParamsBuilder_1 = require("./extraParamsBuilder");
const bigImageCardBuilder_1 = require("./cards/bigImageCardBuilder");
const itemsListCardBuilder_1 = require("./cards/itemsListCardBuilder");
class Reply {
    static text(textDeclaration, extraParamsDeclaration) {
        const textReply = textReplyBuilder_1.TextReplyBuilder.createTextReply(textDeclaration);
        const result = {
            text: textReply.text,
            tts: textReply.tts,
            end_session: false,
        };
        if (extraParamsDeclaration) {
            const extraParams = extraParamsBuilder_1.ExtraParamsBuilder.createExtraParams(extraParamsDeclaration);
            if (extraParams.tts) {
                result.tts = extraParams.tts;
            }
            if (extraParams.buttons) {
                result.buttons = extraParams.buttons;
            }
            if (typeof extraParams.end_session !== 'undefined') {
                result.end_session = extraParams.end_session;
            }
        }
        return result;
    }
    static bigImageCard(textDeclaration, cardDeclaration, extraParamsDeclaration) {
        const textReply = textReplyBuilder_1.TextReplyBuilder.createTextReply(textDeclaration);
        const card = bigImageCardBuilder_1.BigImageCardBuilder.createBigImageCard(cardDeclaration);
        const result = {
            text: textReply.text,
            tts: textReply.tts,
            card: card,
            end_session: false,
        };
        if (extraParamsDeclaration) {
            const extraParams = extraParamsBuilder_1.ExtraParamsBuilder.createExtraParams(extraParamsDeclaration);
            if (extraParams.buttons) {
                result.buttons = extraParams.buttons;
            }
            if (typeof extraParams.end_session !== 'undefined') {
                result.end_session = extraParams.end_session;
            }
        }
        return result;
    }
    static itemsListCard(textDeclaration, cardDeclaration, extraParamsDeclaration) {
        const textReply = textReplyBuilder_1.TextReplyBuilder.createTextReply(textDeclaration);
        const card = itemsListCardBuilder_1.ItemsListCardBuilder.createItemsListCard(cardDeclaration);
        const result = {
            text: textReply.text,
            tts: textReply.tts,
            card: card,
            end_session: false,
        };
        if (extraParamsDeclaration) {
            const extraParams = extraParamsBuilder_1.ExtraParamsBuilder.createExtraParams(extraParamsDeclaration);
            if (extraParams.buttons) {
                result.buttons = extraParams.buttons;
            }
            if (extraParams.end_session) {
                result.end_session = extraParams.end_session;
            }
        }
        return result;
    }
}
exports.Reply = Reply;
