// yandex-dialogs-sdk/dist/reply/cards/bigImageCardBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cardFooterBuilder_1 = require("./cardFooterBuilder");
const cardButtonBuilder_1 = require("./cardButtonBuilder");
class BigImageCardBuilder {
    static createBigImageCard(declaration) {
        if (typeof declaration === 'object') {
            const result = {
                type: 'BigImage',
                image_id: declaration.image_id,
                title: declaration.title,
                description: declaration.description,
            };
            if (declaration.button) {
                result.button = cardButtonBuilder_1.CardButtonBuilder.createCardButton(declaration.button);
            }
            if (declaration.footer) {
                result.footer = cardFooterBuilder_1.CardFooterBuilder.createCardFooter(declaration.footer);
            }
            return result;
        }
        if (typeof declaration === 'string') {
            return {
                type: 'BigImage',
                image_id: declaration,
            };
        }
        throw new Error('Big image card declaration is not of proper type. ' +
            'Could be only string or object.');
    }
}
exports.BigImageCardBuilder = BigImageCardBuilder;
