// yandex-dialogs-sdk/dist/reply/cards/itemsListCardBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cardFooterBuilder_1 = require("./cardFooterBuilder");
const itemsListCardHeaderBuilder_1 = require("./itemsListCardHeaderBuilder");
const itemsListCardImageBuilder_1 = require("./itemsListCardImageBuilder");
function isItemsListCard(declaration) {
    return declaration.items !== undefined;
}
function isItemsListImageArray(declaration) {
    return Array.isArray(declaration);
}
class ItemsListCardBuilder {
    static createItemsListCard(declaration) {
        if (isItemsListImageArray(declaration)) {
            return {
                type: 'ItemsList',
                items: declaration.map(item => itemsListCardImageBuilder_1.ItemsListCardImageBuilder.createItemsListCardImage(item)),
            };
        }
        if (isItemsListCard(declaration)) {
            const result = {
                type: 'ItemsList',
                items: declaration.items.map(item => itemsListCardImageBuilder_1.ItemsListCardImageBuilder.createItemsListCardImage(item)),
            };
            if (declaration.footer) {
                result.footer = cardFooterBuilder_1.CardFooterBuilder.createCardFooter(declaration.footer);
            }
            if (declaration.header) {
                result.header = itemsListCardHeaderBuilder_1.ItemsListCardHeaderBuilder.createItemsListCardHeader(declaration.header);
            }
            return result;
        }
        throw new Error('Card button declaration is not of proper type. ' +
            'Could be only array or object.');
    }
}
exports.ItemsListCardBuilder = ItemsListCardBuilder;
